tests/template_tests/test_engine.py

```diff
@@ -21,6 +22,38 @@ def test_basic_context(self):
         )
 
 
+class GetDefaultTest(SimpleTestCase):
+
+    @override_settings(TEMPLATES=[])
+    def test_raises_when_no_dtl_engines_configured(self):
+        """
+        A configured DTL backend is required.
+        """
+        with self.assertRaisesMessage(ImproperlyConfigured, "No DjangoTemplates backend is configured."):
+            Engine.get_default()
+
+    @override_settings(TEMPLATES=[
+        {
+            'NAME': 'default',
+            'BACKEND': 'django.template.backends.django.DjangoTemplates'
+        },
+        {
+            'NAME': 'extra',
+            'BACKEND': 'django.template.backends.django.DjangoTemplates'
+        },
+    ])
+    def test_multiple_dtl_engines_configured(self):
+        """
+        #27359 -- A DTL backend is returned when several are configured.
+
+        * Users can configure multiple DTL backend (with different options etc).
+        * Third-party libraries still need to be able to call `Template(...)`
+            without passing an explicit engine.
+        """
+        engine = Engine.get_default()
+        self.assertTrue(isinstance(engine, Engine))
```

Comment:
The only way to check the identity of the `engine` is to re-perform the exact filtering logic from `get_default`, which—being the method under test—wouldn't add anything. (Except copy/paste maintenance overhead.)

Prior to this PR, there wasn't explicit coverage on this method. Both branches are now tested. 

The _first_ result is just the behaviour from `list`—which presumably we can rely on. 

On the principle of _"Obviously no errors, or no errors that are obvious"_, I'd say the implementation is simple enough not to need more checking. 

We can add the extra check if it's insisted upon, but I think it better without.